src/plans/stepsize.jl

```diff
@@ -756,6 +757,64 @@ function show(io::IO, a::NonmonotoneLinesearch)
 end
 get_message(a::NonmonotoneLinesearch) = a.message
 
+@doc raw"""
+    PolyakStepsize <: Linesearch
+
+
+Compute a step size
+
+````math
+α_i = \frac{f(p^{(k)}) - f_{\text{best}}+γ_k}{\lVert{ ∂f(p^{(k)})} \rVert^2},
+````
+
+where ``f_{\text{best}}`` is the (up to now) best cost function seen until now,
+and ``γ_i`` is a sequence of numbers that is square summable, but not summable.
+Furthermore ``∂f`` denotes a subgradient of ``f`` at the current iterate ``p^{(k)}``.
+
+The step size is an adaption from the Dynamic step size discussed in Section 3.2 of [Bertsekas:2015](@cite),
+both to the Riemannian case and to approximate the minimum cost value.
+
+# Fields
+
+* `γ`               : a function `i -> ...` representing the sequence.
+* `best_cost_value` : storing the value `f_{\text{best}}`
+
+# Constructor
+
+    PolyakStepsize(;
+        γ = i -> 1/i,
+        initial_cost_estimate=0.0
+    )
+
+initialize the Polyak stepsize to a certain sequence and an initial estimate of ``f_{\text{best}}``
+"""
+mutable struct PolyakStepsize{F,R} <: Linesearch
+    γ::F
+    best_cost_value::R
+end
+function PolyakStepsize(; γ::F=(i) -> 1 / i, initial_cost_estimate::R=0.0) where {F,R}
+    return PolyakStepsize{F,R}(γ, initial_cost_estimate)
+end
+function (ps::PolyakStepsize)(
+    amp::AbstractManoptProblem, ams::AbstractManoptSolverState, i::Int, args...; kwargs...
+)
+    # We get these by reference, so that should not allocate in general
+    M = get_manifold(amp)
+    p = get_iterate(ams)
+    X = get_subgradient(amp, p)
+    # Evaluate the cost
+    c = get_cost(M, get_objective(amp), p)
+    (c < ps.best_cost_value) && (ps.best_cost_value = c)
+    α = (c - ps.best_cost_value + ps.γ(i)) / (norm(M, p, X)^2)
+    return α
+end
+function show(io::IO, ps::PolyakStepsize)
+    return print(
+        io,
+        "PolyakStepsize() with keyword parameters\n  * initial_cost_estimate = $(ps.best_cost_value)",
```

Comment:
`best_cost_value` may be different from `initial_cost_estimate` so this could be misleading. Maybe let's either call it `best_cost_value` or not print it at all.

Reply:
If you want to start from the time printing you have to set that number as the initial one, that's why I printed it as such (also to reflect the keyword).

Reply:
But sure we could also not print it if you prefer, I would prefer to provide that information as the current state.

Reply:
I see, then maybe let's keep it as is.